BoneReconstructionPlanner/BoneReconstructionPlanner.py

```diff
@@ -324,10 +365,8 @@ def onAddCutPlaneButton(self):
     slicer.mrmlScene.AddNode(planeNode)

```

Comment:
Move this method to logic.

Reply:
What do I do with this line inside that function?
`self.planeNodeObserver = planeNode.AddObserver(slicer.vtkMRMLMarkupsNode.PointPositionDefinedEvent,self.onPlanePointAdded)
`
The member variable is used to later delete the observer in _onPlanePointAdded_
If I move that function to the logic, I should move _onPlanePointAdded_ and _onPlaneModified_ also

Reply:
Yes, the logic should observer all cutting plane nodes and whenever any of them are changed then do all the necessary updates. 

GUI widget should do trivial operations when an observed node changes, for example update the widget content or button states. Calling logic update functions form the GUI in response to node changes should be usually avoided (automatic updates are better to do in the logic).

Think about the widget class as an optional component. The module should be usable just by calling logic functions. It can help (and it is also useful for software quality assurance) if you implement a few automatic tests. Since most automatic tests only uses logic functions, it will be clear if the logic is self-contained enough.

Reply:
Look at this line from _onPlaneModified_:
`self.logic.generateFibulaPlanes(self.ui.fibulaLineSelector.currentNode(), self.ui.mandibleCurveSelector.currentNode(), mandibularPlanesList, self.initialSpace, self.betweenSpace, self.ui.rightFibulaCheckBox.isChecked())
`
I need those parameters from the widget, should I save them on the Parameter Node so I can access them in the logic?

Reply:
Yes, of course, widget immediately writes all parameter changes into the parameter node. The logic observes the parameter node and performs all necessary updates accordingly.